### PR TITLE
Fix issue 6 by proxying doc, dict and bases

### PR DESCRIPTION
**Pull request by jamadden**
Spread across a few commits for clarity. 

There's some whitespace cleanup in the C file in its own commit (there were mixed tabs and spaces, making it difficult to read); ?w=0 might be useful to see the real changes.

### CODE REVIEW

File: src/zope/hookable/_zope_hookable.c [hunk 1]
```diff
@@ -143,10 +127,57 @@ hookable_call(hookable *self, PyObject *args, PyObject *kw)
...
+    name_as_string = PyString_AsString(name);
+#else
+    name_as_string = PyUnicode_AsUTF8(name);
+#endif
+
+    if (name_as_string == NULL) {
+        return NULL;
+    }
+
+    maybe_special_name = strlen(name_as_string) > 2 && name_as_string[0] == '_' && name_as_string[1] == '_';
```
[the remark this answers is not in the document]
Comment (mgedmin) [reply src/zope/hookable/_zope_hookable.c]: Heh, I just noticed that zope.proxy uses exactly that: https://github.com/zopefoundation/zope.proxy/pull/25/files#diff-48220ec482a37b43d11f14b9e3f2d807R255

[the remark this answers is not in the document]
Comment (jamadden) [reply src/zope/hookable/_zope_hookable.c]: Yeah, that's where I initially started to copy this code from 😄 

It struck me as unsafe that it didn't do a strlen, but clearly I hadn't thought it through. Thanks for the observation! I'll make a change.

File: src/zope/hookable/__init__.py [hunk 2]
```diff
@@ -18,16 +18,33 @@ class _py_hookable(object):
...
 
     def __init__(self, *args, **kw):
-        if len(args) == 0 and 'implementation' in kw:
+        if not args and 'implementation' in kw:
             args = (kw.pop('implementation'),)
         if kw:
             raise TypeError('Unknown keyword arguments')
         if len(args) != 1:
             raise TypeError('Exactly one argument required')
         self._original = self._implementation = args[0]
```
[the remark this answers is not in the document]
Comment (jamadden) [reply src/zope/hookable/__init__.py]: I think it's meant to be bug-for-bug compatible with the C implementation (every line is covered by a test), which uses `PyArg_ParseTupleAndKeywords(args, kwds, "O:hookable", {"implementation", NULL}, ...)`.

**Comment by jamadden** [comment]
Thank you!

**Pull request merged**